### PR TITLE
[chore] move log to registry

### PR DESCRIPTION
**Pull request by trisch-me**
While moving log attributes to the registry I have noticed that we have a repetition of `file` namespace for some of the log attributes. I have replaced them with fields from `file` namespace and deprecated old attributes.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: schema-next.yaml [hunk 1]
```diff
@@ -2,6 +2,13 @@ file_format: 1.1.0
 schema_url: https://opentelemetry.io/schemas/next
 versions:
   next:
+    logs:
+      changes:
+        # https://github.com/open-telemetry/semantic-conventions/pull/908
+        - rename_attributes:
+            attribute_map:
+              log.file.name: file.name
```
Comment (ChrsMark) [inline schema-next.yaml:10]: FYI @djaglowski we would need to inherit this breaking change in the filelog receiver as well. Most probably through a feature flag? Would that be doable?

**Review by ChrsMark — COMMENTED** [review]
@trisch-me do we know the impact of this breaking change (log.file.path -> file.path)?

I know for example that this would affect the `recombine` parser for [k8s container logs](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/50098347046c7558788de0100cd5e72f5e38fdac/charts/opentelemetry-collector/templates/_config.tpl#L229).

Shall we collect all the possible affected implementations and evaluate the impact before merging this one?

**Comment by ChrsMark** [comment]
A search within the whole org give some additional impacted projects: https://github.com/search?q=org%3Aopen-telemetry+log.file.name&type=code&p=1.

Most of them are language specific implementations. 
Similar for [`log.file.path`](https://github.com/search?q=org%3Aopen-telemetry+log.file.path&type=code&p=2).

In particular [k8s container logs](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/50098347046c7558788de0100cd5e72f5e38fdac/charts/opentelemetry-collector/templates/_config.tpl#L229) would break all users collecting container logs (in cri-o format) from k8s using Opentelemetry's helm charts presets. @TylerHelmuth feel free to correct me here and if you have any ideas of how this should be handled by the helm charts project.

I'm not opposing to that change, just want to ensure that we will properly handle it.

**Review by djaglowski — CHANGES_REQUESTED** [review]
I'm not convinced this change is necessary or correct.

The intention behind the `log.*` namespace is to convey information about the mechanism of log transmission, which may need to exist simultaneously with other attributes describing a file.

For example, an application which tries to open a file and fails may emit a structured log e.g. 
```yaml
level: error
message: failed to open file
file.name: foo.csv
time: ...
```

In such a case, the purpose of `log.file.name` is distinct from `file.name`. I think we should reserve this dedicated namespace for the purpose of describing log media separately from log content.

**Comment by trisch-me** [comment]
@djaglowski I have a question: do you think there might be in this structured log information about some other file? If this comes in one message, information is connected, so `file` attributes will be connected to the `log`.

Is it different from your point of view?

**Comment by djaglowski** [comment]
> do you think there might be in this structured log information about some other file?

Yes, in the example above I'm trying to show that an application which needs to access `foo.csv` may write a log describing this interaction to a log file which is called something else, say `application.log`. Both of these file names have a distinct semantic meaning - one to describe the application event, the other to describe how the event was transmitted.

**Comment by trisch-me** [comment]
Now I got your point. I agree with you in this case it will be different files. This will be resolved as soon as we will implement re-usage of the namespaces within other namespaces, so in that case we will have `file.*` on the top level and `log.file.*` both pointing to the same file namespace. But for now I will revert this change and will add a TODO remark

**Comment by trisch-me** [comment]
I have reverted changes, please have a look

File: model/registry/log.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            - id: stdout
+              value: 'stdout'
+              brief: 'Logs from stdout stream'
+              stability: experimental
+            - id: stderr
+              value: 'stderr'
+              brief: 'Events from stderr stream'
+              stability: experimental
+
+  - id: registry.log.file    # TODO: should we move it to the file model?
```
Comment (ChrsMark) [inline model/registry/log.yaml:24]: Would that make sense to move this under `registry.log.*` namespace here in the registry? In order to honor the "one single and flat registry" idea.

File: schema-next.yaml (hunk 1 above)
Comment (ChrsMark) [reply schema-next.yaml]: I think that one is resolved now after the conclusion at https://github.com/open-telemetry/semantic-conventions/pull/908#issuecomment-2059918201.
Thank's @djaglowski for the input here!

File: model/registry/log.yaml (hunk 2 above)
Comment (trisch-me) [reply model/registry/log.yaml:24]: We don't have agreement on this [yet](https://github.com/open-telemetry/semantic-conventions/issues/887).

Comment (trisch-me) [reply model/registry/log.yaml:24]: We can proceed and change then later as soon as there will be any decision made. No decision is also a decision :)

**Review by joaopgrassi — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Pull request merged**